### ISSUE TITLE
Better veritical text recognition 

### ISSUE THREAD

**Issue by codeguruuu** [matt-m-o/YomiNinja#14]
I've played around with this tool a bit and ive noticed it's got alot of potential, ive tried games and for the most part it worked flawlessly even **OUTSIDE** of games because its an ocr litteraly anything with text can be mined. Which i think that's what gives this tool so much potential However, I've noticed **vertical text** gives me a weird problem. The ocr seems to get confused when it comes to vertical text and its kinda buggy so i think that should be fixed. But thank you for the amazing tool you're doing an amazing job!

**Comment by matt-m-o** [comment]
Thank you for your kind words and feedback! I've noticed some challenges, especially when dealing with manga and similar formats. To better address the issue, could you please provide an example of a problematic result? This will help me determine whether the issue arises from the OCR process itself or from the subsequent presentation of the results.
If possible, also provide the original image (without the overlay) so I can use it for testing.

**Comment by codeguruuu** [comment]
Yes ofc, for this i just grabbed a random manga but it can get alot worse with alot of text
![YomiNinja_WCyS6u4X7s](https://github.com/matt-m-o/YomiNinja/assets/129904858/a386866d-68b9-4ce3-8db9-3eb92bfadd3f)

as you can see the ocr is getting confused and is trying to go line by line like horizontal text and the ocr boxes are overlapping. This makes it much more harder to mine. But if your determined enough you can mine it. However this is only with these types. The others you couldn't even if you wanted to, It would of been much more effective if it just used one box

To show an example of just how bad it can get with more text, Check this one down below

![image](https://github.com/matt-m-o/YomiNinja/assets/129904858/bff8945f-cd94-48b8-b2df-44b3074b1396)

These are unmineable for the most part especially **bottom right** I've provided the orginal images on imgur right here 
[https://imgur.com/a/gSkl4Hv](url) I hope this helps with making the tool better!

**Comment by matt-m-o** [comment]
Your examples were very helpful. Thank you for sharing!
I have made some improvements to the overlay, and the current results are as follows:

![yn 0 6 0 Improved Vertical Text 1](https://github.com/matt-m-o/YomiNinja/assets/25914763/cd63539c-2c82-4e3b-aa95-1520ecc337b7)

![yn 0 6 0 Improved Vertical Text 2](https://github.com/matt-m-o/YomiNinja/assets/25914763/97cb4492-2826-41ac-b054-1250c1e8244f)


I still don't think it is currently suitable for manga, as further improvements are still needed for the current OCR engine. Once additional OCR engines are integrated, which will happen very soon, the results should be significantly better.

**Comment by codeguruuu** [comment]
Wow, while not a 100% perfect i think thats a significate improvement from before and yeah i agree. i will be waiting to see what you got in plan for the future!

**Comment by matt-m-o** [comment]
**Update:** The current v0.7.2 offers better vertical text recognition. However, significant improvements are coming in v0.8.

**[v0.7.2](https://github.com/matt-m-o/YomiNinja/releases/tag/v0.7.2) - [Manga OCR](https://github.com/kha-white/manga-ocr)+ Paddle Text Detector**
![](https://github.com/user-attachments/assets/16bcd224-0f59-49d7-9950-4da1f7a6b661)


**v0.8 - [Manga OCR](https://github.com/kha-white/manga-ocr) + [Comic Text Detector](https://github.com/dmMaze/comic-text-detector)**
![](https://github.com/user-attachments/assets/486241a8-e498-45a7-b8a0-7eda361c4227)


**[v0.7.2](https://github.com/matt-m-o/YomiNinja/releases/tag/v0.7.2) - Google Lens**
![Before Manga GoogleLens](https://github.com/user-attachments/assets/7c71ff9d-ecb3-4bce-9b26-e962d87a887d)

**v0.8 - Google Lens**
![After Manga Google Lens](https://github.com/user-attachments/assets/be1522a4-3cfc-425a-8360-08044e0d1a32)
